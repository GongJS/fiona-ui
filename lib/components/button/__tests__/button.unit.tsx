import React from 'react';
import Button from '../button';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('Button Component', () => {
  it('renders button component', () => {
    const component = renderer.create(<Button>Primary</Button>);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('button component click events', () => {
    const component = renderer.create(<Button onClick={() => {}}>Primary</Button>);
    const json = component.toJSON();
    json && json.props.onClick();
  });
  it('button component text', () => {
    const button = shallow(<Button>Primary</Button>);
    expect(button.text()).toEqual('Primary');
  });
});