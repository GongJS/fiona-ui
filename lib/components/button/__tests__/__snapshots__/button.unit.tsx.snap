// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders button component 1`] = `
<button
  className="f-button f-button-default"
  onClick={[Function]}
>
  <span
    className=""
  >
    Primary
  </span>
</button>
`;
